modules/nextflow/src/main/groovy/nextflow/processor/PublishDir.groovy

```diff
@@ -364,6 +369,11 @@ class PublishDir {
     @CompileStatic
     protected void processFile( Path source, Path destination ) {
 
+        // resolve Fusion symlink if applicable
+        if( FusionHelper.isFusionEnabled(Global.session as Session) )
+            if( source.name in taskInputs )
```

Comment:
What's the semantics for `in` operator against a map? does it check it via `.containsKey`?

Reply:
Apparently the `in` operator casts the corresponding value to a boolean, which means e.g. `'foo' in ['foo': null]` would return false:

https://www.baeldung.com/groovy-collections-find-elements#3-map

Aside from that it is the same as `containsKey`

Reply:
Under the hood, the [membership operator](https://groovy-lang.org/operators.html#_membership_operator) is semantic sugar for [.isCase()](https://docs.groovy-lang.org/latest/html/groovy-jdk/java/util/Map.html#isCase(java.lang.Object)) for most objects (including Maps), and [.contains()](https://docs.groovy-lang.org/latest/html/groovy-jdk/java/lang/Iterable.html#contains(java.lang.Object)) for Lists.

The semantics are exactly as you describe Ben.